operators/src/source/gdal_source.rs

```diff
@@ -370,6 +370,84 @@ impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
     }
 }
 
+#[derive(PartialEq, Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct GdalMetadataFixedTimes {
+    pub time_steps: Vec<TimeInterval>,
+    pub minimum_step: TimeStep,
+    pub time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+    pub params: GdalDatasetParameters,
+    pub result_descriptor: RasterResultDescriptor,
+}
+
+#[async_trait]
+impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for GdalMetadataFixedTimes
+{
+    async fn loading_info(&self, query: RasterQueryRectangle) -> Result<GdalLoadingInfo> {
+        let valid_duration = TimeInterval::new(
+            self.time_steps[0].start(),
+            self.time_steps[self.time_steps.len() - 1].start(),
+        )?;
+
+        let parts = match valid_duration.intersect(&query.time_interval) {
+            Some(time_interval) => {
+                let mut loading_info_parts: Vec<GdalLoadingInfoPart> = Vec::new();
+
+                for (index, time_position) in self.time_steps.iter().enumerate() {
+                    if time_interval.contains(time_position) {
+                        loading_info_parts.push(
+                            self.params
+                                .replace_time_placeholders(&self.time_placeholders, *time_position)
+                                .map(|loading_info_part_params| GdalLoadingInfoPart {
+                                    time: *time_position,
+                                    params: loading_info_part_params,
+                                })?,
+                        );
+
+                        if index < self.time_steps.len() - 1
+                            && time_interval.contains(&self.time_steps[index + 1])
+                        {
+                            let time_gap = TimeInterval::new(
+                                (time_position.end() + self.minimum_step)?,
+                                (self.time_steps[index + 1].start() - self.minimum_step)?,
+                            )?;
+
+                            loading_info_parts.push(
+                                self.params
+                                    .replace_time_placeholders(&self.time_placeholders, time_gap)
+                                    .map(|loading_info_part_params| GdalLoadingInfoPart {
+                                        time: time_gap,
+                                        params: loading_info_part_params,
+                                    })?,
+                            );
+                        }
+                    }
+                }
+
+                loading_info_parts
+            }
+            .into_iter(),
+
+            None => vec![].into_iter(),
```

Comment:
the result shouldn't be empty. It should be one single loading info that's valid for the entire `query.time_interval` and should produce nodata tiles. Just use the `replace_time_placeholders` with the `query.time_interval` here. When the gdal source tries to load the data it will find nothing and produce no data.